official/mnist/mnist_tpu.py

```diff
@@ -168,6 +186,19 @@ def main(argv):
   if FLAGS.eval_steps:
     estimator.evaluate(input_fn=eval_input_fn, steps=FLAGS.eval_steps)
 
+  # Run prediction on the test data.
+  if FLAGS.enable_predict:
+    predictions = estimator.predict(input_fn=predict_input_fn)
+
+    for pred_dict in predictions:
+      template = ('Prediction is "{}" ({:.1f}%), expected "{}"')
```

Comment:
In general, isn't `predict` typically used to apply the network on previously unseen data? So there won't be an "expected" answer. Since this PR is intended to provide an example, I'd leave out the "expected" business (which only happens to be available since we're using the test set). Or better still, `predict_input_fn` shouldn't even produce the "expected" tensor (as that would be the case when using `predict` in practice, no?)

(See for example: https://www.tensorflow.org/api_docs/python/tf/contrib/tpu/TPUEstimator#example_mnist_2)

Reply:
Done. Could not remove the expected tensor from the dataset as the test() function is generating zipped dataset with features & labels. And I didn't find a suitable way to separate those out.
But as we are not using labels anywhere in the prediction call stack so it won't be a problem.